buildkite/resource_pipeline_test.go

```diff
@@ -109,6 +189,20 @@ func TestAccPipeline_add_remove_withcluster(t *testing.T) {
 					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "allow_rebuilds", "true"),
 				),
 			},
+			{
+				Config:                   testAccPipelineConfigBasic("foo"),
+				ProtoV6ProviderFactories: protoV6ProviderFactories(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the pipeline exists in the buildkite API
+					testAccCheckPipelineExists("buildkite_pipeline.foobar", &resourcePipeline),
+					// Confirm the pipeline has the correct values in Buildkite's system
+					testAccCheckPipelineRemoteValues(&resourcePipeline, "Test Pipeline foo"),
+					// Confirm the pipeline has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "name", "Test Pipeline foo"),
+					resource.TestCheckNoResourceAttr("buildkite_pipeline.foobar", "cluster_id"),
```

Comment:
You can see the state change here. In the previous versions of the provider, if no cluster is attached to the pipeline, the `cluster_id=""` in state. But in this version of the provider it is set to null. Its effectively the same and upon a `terraform refresh` the state will be updated under-the-hood for users

Reply:
s above with my commentary 👍